.github/workflows/python-version-tests.yml

```diff
@@ -0,0 +1,35 @@
+name: Python Version Tests
+
+on:
+  push:
+    branches: [ "master" ]
+  pull_request:
+  workflow_dispatch:
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        python-version: ["3.5", "3.6", "3.7", "3.8"]

```

Comment:
you can probably get rid of `3.5` and `3.6`, since they are no longer supported.  It may also be worth adding `3.9`.  I believe there is a bigger change with Python `3.10`, so it's probably not worth taking on in this PR.